Silicon/Bosc/NanHuPkg/Library/PciSegmentLib/PciSegmentLib.c

```diff
@@ -0,0 +1,1382 @@
+/** @file
+  PCI Segment Library for SoC with multiple RCs.

```

Comment:
Is there a SoC specific behavior here? It looks all generic library code. So, can this be added under RISC-V itself so that any RISC-V SoC can make use of it instead of duplicating? Much better if edk2 library can be used.